modules/caddyhttp/fileserver/browse.go

```diff
@@ -49,8 +49,9 @@ func (fsrv *FileServer) serveBrowse(root, dirPath string, w http.ResponseWriter,
 	// of "/a/b/c" - so we have to redirect in this case
 	oldReq := r.Context().Value(caddyhttp.OriginalRequestCtxKey).(http.Request)
 	if !strings.HasSuffix(oldReq.URL.Path, "/") {
+		oldReq.URL.Path += "/"
 		fsrv.logger.Debug("redirecting to trailing slash to preserve hrefs", zap.String("request_path", oldReq.URL.Path))
```

Comment:
I think we might want to do the append second (i.e. switch these two lines, as in the original) -- so we log which path is being redirected from.